maven/WFLY-14934_Server_Plugin.adoc

```diff
@@ -0,0 +1,213 @@
+
+A Galleon configuration is expected, if no galleon configuration is found, the goal execution fails.
+
+The configuration items names, when applicable, are re-using the names that are in use in existing WildFly Maven plugin goals.
+
+* packaging-scripts: A list of execution of CLI scripts and commands. Each execution can contain: a list of CLI scripts, a list of commands, a list of properties files, 
+a list of java options, the ability to resolve or not expressions located in CLI scripts locally (false by default) and system properties for the CLI forked process. 
+NB: CLI commands are automatically executed in the context of an embedded server.
+* extra-server-content-dirs: A list of directories to copy content to the provisioned server inside a forked process. 
+The content is copied into the provisioned server keeping the directory structure found inside 
+the extra directories. e.g.: ```my-extra-dir/standalone/configuration/foo.properties```, the file ```foo.properties``` will 
+be located inside the ```<provisioned server>/standalone/configuration/foo.properties```.
+* filename: The name of the artifact to deploy. A look-up is done inside the project build directory. 
+By default a lookup is done to retrieve the primary artifact to deploy. If no file is found, no deployment is operated.
+* name: The name of the deployment. By default the file name is used.
+* runtime-name: The runtime name of the deployment. By default the file name is used. NB: In order to deploy 
+the deployment inside the root context, ```ROOT.war``` runtime name is to be used.
+* server-config: The server config to use when executing CLI scripts and deploying artifact. Is ```standalone.xml``` by default.
+* stdout: Indicates how ```stdout``` and ```stderr``` should be handled for the forked CLI process. 
+By default the output is redirected to the current process. ```none``` to disable output, ```System.out``` or ```System.err``` to redirect to the current process. 
+Any other value is assumed to be the path to a file and the out/err will be written there. All CLI executions output is appended to the file.
+
+==== ```run``` and ```start``` goals impact
+
+* If jboss-home has not been set and no server if found in ```<buildDir>/<provisioning-dir>``` a server is provisioned.
+* The ``artifact``, ``groupId``, ``artifactId``, ``classifier``, ``packaging`` configuration items have been removed.
+* The ```version``` configuration item is kept allowing to identify a WildFly version to install.
+* The ```provisioning-dir``` allows to specify where to (find/provision) the server.
+
+==== ```execute-commands``` goal impact
+
+* The ability to provision a server if no server installation is found is removed.
+
+==== WildFly Maven plugin versioning
+
+A new major version would contain these new features.
+
+==== Impact on quickstarts 
+
+* A set of quickstarts (List TBD) will be refactored to use this plugin in both bare-metal and cloud.
```

Comment:
Can this TBD be resolved?

Reply:
Not in the context of RFE.

Reply:
@jmesnil Thank you. What I meant is we shouldn't approve analysis including TODOs/TBDs. If quickstarts are not subject of this proposal, it should be IMHO included.

Reply:
@jfdenise actually, as the first release focuses on cloud deployment, we should not mention that it will also be use for bare-metal.
Can you also please remove the `(List TBD)` as the list of refactored quickstarts will not be defined by this RFE. We will define that lists when we have S2I images that leverages this plugin.

Reply:
Ok, yes. Fixed.